### PR TITLE
Replace dynamic invoke

### PR DESCRIPTION
**Pull request by robinsedlaczek**
Changed the invocation process of extension function to gain better performance. Performance of SharpGL has been increased by about 30% (on my machine).

### CODE REVIEW

**Comment by dwmkerr** [comment]
Hi Robin,

Right, let's do this! Can you change the target of the pull request from `master` to `v/2.4`, I'd like to merge this directly into 2.4, but then make a few small changes. I've put some comments on the PR, I'd also like to run the performance tests myself and have a play with them. I think it will be worth documenting these changes somehow. 

I also wonder whether we should create a page on the wiki talking about SharpGL performance. This is  because others may find our learnings on improving the performance of a library like this very useful.

Finally, I'm also going to try and get metrics from the RedGate profiler so that I can show some pretty graphs in the release note showing how the performance has improved :)

File: source/SharpGL/Core/SharpGL/RenderContextProviders/DIBSectionRenderContextProvider.cs [hunk 1]
```diff
@@ -84,15 +84,18 @@ public override void SetDimensions(int width, int height)
 
 	    public override void Blit(IntPtr hdc) 
 	    {
+            // [RS] Because of this warning, we remove the if-statement.
+            // Warning 1: The result of the expression is always 'true' since a value of type 'System.IntPtr' is never equal to 'null' of type 'System.IntPtr?'	F:\Git Repositories\SharpGL\source\SharpGL\Core\SharpGL\RenderContextProviders\DIBSectionRenderContextProvider.cs	88	17	SharpGL
+
             //  We must have a device context.
-            if (deviceContextHandle != null)
```
Comment (dwmkerr) [inline source/SharpGL/Core/SharpGL/RenderContextProviders/DIBSectionRenderContextProvider.cs]: Correct, this code is totally wrong. What it should be is:

```cs
if (deviceContextHandle == IntPtr.Zero)
{
    return;
}
// not 
// if (deviceContextHandle != null)
```

As there is no point swapping buffers if the dc is null. Also, inverting the logic means less indentation for the 'real work' of the code

Comment (robinsedlaczek) [inline source/SharpGL/Core/SharpGL/RenderContextProviders/DIBSectionRenderContextProvider.cs]: Perfect! I like this style! :) So I'll do that and send it via PR targeted to 2.4.

**Comment by robinsedlaczek** [comment]
Hey Dave,
I created a Wiki page. Hope it is ok in this style. I'll document the change with motivation, change and results later today. I'll send the PR to 2.4 later today as well.
Kind regards!
Robin

___________________________________Robin SedlaczekWave Developments / The Berlin Microsoft Connection Warschauer Strasse 6010243 BerlinGermany Twitter https://twitter.com/RobinSedlaczekWordpress http://robinsedlaczek.wordpress.com/Facebook https://www.facebook.com/TheBerlinMicrosoftConnectionLinkedIn http://de.linkedin.com/pub/robin-sedlaczek/34/6a6/222/Xing https://www.xing.com/profile/Robin_Sedlaczek


Date: Tue, 20 Jan 2015 00:16:29 -0800

**Comment by dwmkerr** [comment]
Thanks Robin, looking forward to getting this change in, it's been far too long! That's fine to create a wiki page, go right ahead!

**Comment by dwmkerr** [comment]
Replaced with new PR

**Pull request closed without merging**